.github/workflows/device-tests-android.yml

```
@@ -110,10 +112,33 @@ jobs:
           script: pwsh scripts/device-test.ps1 android -Run -Tfm ${{ matrix.tfm }}
 
       - name: Retry Tests (if previous failed to run)
-        if: steps.first-run.outcome == 'failure'
+        if: steps.first-test-run.outcome == 'failure'
         timeout-minutes: 40
         uses: reactivecircus/android-emulator-runner@1dcd0090116d15e7c562f8db72807de5e036a4ed # Tag: v2.34.0
+        with: *android-emulator-test
+
+      - name: Setup Environment
+        uses: ./.github/actions/environment
+
+      - name: Select Java 17
+        uses: actions/setup-java@dded0888837ed1f317902acf8a20df0ad188d165 # v5.0.0
+        with:
+          distribution: ${{ runner.os == 'Windows' && runner.arch == 'ARM64' && 'microsoft' || 'temurin' }}
+          java-version: '17'
+
+      - name: Checkout github-workflows
+        uses: actions/checkout@v5
```

```suggestion
        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
```

considering https://github.com/getsentry/sentry-dotnet/pull/4562
I believe this is the only new `uses:` without being pinned to a SHA